Assignment 5.Rmd

```diff
@@ -40,42 +75,90 @@ We want to see if we can build a decision tree to help teachers decide which stu
+
+score_ctree <- ctree(factor(advice) ~ prior_prob_count +
+                                      prior_percent_correct + 
+                                      hints,
+                     data = D1)
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree)
 ```
 
-Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
+Please interpret the tree, which two behaviors do you think the teacher should most closely pay attention to?
+
+*The output does not show properly - bar on the left is monitor, middle is intervene, and right is no action. This can be seen if you open the plot in a new window and maximise it.*
+
+*Teachers should definitely pay attention to hints. Hints are relevant at  two split levels. It also serves as an initial split. Those who ask for any hints (right of the tree) are far more likely to be in the intervene and monitor groups, while those who ask for no hints are much more likely to require no action. Those who ask for many hints probably require more intervention than those who ask for less as well.* 
+
+*Teachers should also pay attention to prior prob count correct. Relative to prior percent correct, differences in prior prob count are decisive for more students (240 vs 131), suggest greater magnitude increases in the number of students where teachers need to intervene rather than take no action, and similar magnitude increases in the proportion that they should monitor rather than take no action*
 
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv")
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
-
-D2$prediction <- 
+D2$prediction <- predict(score_ctree, D2)
 
 ``` 
+
+
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
 
+```{r}
+#Compute the actual advice that the Test set students received
+#Weirdly all the score data for the test set == 1, so we should expect everyone in the test set to require no action.
+D2$advice <- ifelse(D2$score > 0.3,
+                    ifelse(D2$score > 0.7,
+                           "no_action",
+                           "monitor"),
+                    "intervene"
+                    )
+
+#Compute difference between observed and predicted results
+conf_mat <- table(D2$advice, D2$prediction)
+conf_mat
+
+#Calculate accuracy based on take no action being the 'positive' result - did the model correctly identify those who require no action?
+accuracy <- conf_mat[1,3]/sum(conf_mat[1, ])
+accuracy
+
```

Comment:
You can also include the calculation of the error rate the following way: `Errorrate  <- 1-accuracy`.